### DISCUSSION TITLE
Loading screen on AR

### DISCUSSION THREAD

**Discussion by sercandemirtas** [google/model-viewer#2214] [Q&A]
Hello all!

Is it possible to add loading gif or picture to default AR button on mobile? I mean when we click AR, sometimes we need to wait 3 or 4 second. So I want to add something like "Please wait. AR is loading.."

### Version

- model-viewer: v1.1.0

### Browser Affected

- [x ] Chrome
- [ ] Edge
- [ ] Firefox
- [ x] IE
- [x ] Safari

### OS

- [ x] Android
- [ x] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

**Comment by elalish** [comment]
I've noticed this annoying lag too, especially on iOS. Yet another reason I'll be excited when WebXR is supported universally. We don't expose anything like this yet, but I might be able to add a value to our `ar-status` attribute that you could use for configuring some CSS. Does that sound like a reasonable solution?

**Comment by sayinserdar** [comment]
I had the same problem. My solution was just placing an overlay loader before the model loads

![Loader](https://user-images.githubusercontent.com/26919126/89714446-6b3ac180-d9a7-11ea-9712-7a44979cd60e.gif)

Here is how I did it and I think it would be really helpful to see more advanced/more usage of slot attribute other than button element. In this case I've used "poster" slot.

<img width="646" alt="Screen Shot 2020-08-08 at 6 48 26 PM" src="https://user-images.githubusercontent.com/26919126/89714481-c371c380-d9a7-11ea-98c8-3b17edb5cbfa.png">
<img width="585" alt="Screen Shot 2020-08-08 at 6 47 52 PM" src="https://user-images.githubusercontent.com/26919126/89714485-c5d41d80-d9a7-11ea-8c59-4437982cff5a.png">

You can check it out from this link as well.
[Example](https://app.diner.com.tr/house-of-b/menu/5ed2a2f733455600af8cce80)

**Comment by milesgreen** [comment]
@elalish - I've noticed this too. When tapping the AR button, it can pause / hang / load for a few seconds before opening Quick Look on iOS. Seems intermittent. Possibly depending on connection speed of mobile?  Is the USDZ being loaded at that point?

Either way, if this can be exposed as an ar-status event, that would be great. But would you be able to also provide an event when exiting Quick Look / Scene Viewer, returning back to the web page - to remove the loading CSS?

**Comment by elalish** [comment]
@milesgreen yes, that's when the `ios-src` gets downloaded. Unfortunately I'm not sure I can fix this after all, as it appears the page is unresponsive as soon as the intent is launched on my iphone. If anyone knows of a website that launches quick-look and manages to avoid this delay or put in a loading indicator, please paste a link and I'll try to adapt it.
